### PR TITLE
Add darkmode

### LINKED ISSUE #5: Add dark mode
Like in Dan's https://overreacted.io/

### PR DESCRIPTION
**Pull request by mtliendo**
closes #5 

PR is up in case anyone wants to check it out.

Submitting as a draft so I can get some further direction on color palette, but functionality should all be in place 😄

### CODE REVIEW

**Comment by lipis** [comment]
would be nice to use the same pallet as we use in the app: https://yeun.github.io/open-color/ but how exactly it should look like I have no clue :)

**Comment by mtliendo** [comment]
Oh nice, I have to call it a night for now, but I can fiddle with the colors based on the palette and put them in [what looks to be](https://github.com/gaearon/overreacted.io/blob/master/src/utils/global.css#L1-L32) the right spots.

**Comment by lipis** [comment]
Instead of the checkbox, can we have something like in Dan's blog?

**Comment by mtliendo** [comment]
For sure, that was just placeholder. I’ll be sure to throw in a slider.

File: package.json [hunk 1]
```diff
@@ -3,6 +3,7 @@
   "dependencies": {
     "gatsby": "2.20.2",
     "gatsby-image": "2.3.0",
+    "gatsby-plugin-dark-mode": "^1.1.1",
```
Comment (lipis) [inline package.json]: pin dependencies

**Comment by mtliendo** [comment]
Ready for second pass :)

**Comment by dwelle** [comment]
The commit history is in a bad state right now (duplicating the posts and all). Try to resolve it, or I can look at it later myself.

**Comment by lipis** [comment]
Can we put the toggle in the top right corner? `fixed` position maybe?

File: src/components/layout.js [hunk 2]
```diff
@@ -15,6 +22,46 @@ const Layout = ({ location, title, children }) => {
         padding: `${rhythm(1.5)} ${rhythm(3 / 4)}`,
       }}
     >
+      <div
+        style={{
+          display: "flex",
+          justifyContent: "flex-end",
+          marginBottom: "1.2em",
```
Comment (lipis) [inline src/components/layout.js]:
```suggestion
          position: "fixed",
          right: "8px",
          top: "8px",
```

File: src/components/layoutStyles.css [hunk 3]
```diff
@@ -0,0 +1,40 @@
+/* global.css */
+
+body {
+  --bg: white;
```
Comment (lipis) [inline src/components/layoutStyles.css]: #fff

File: src/components/layoutStyles.css [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  --textNormal: #222;
+  --textTitle: #222;
+  --textLink: blue;
```
Comment (lipis) [inline src/components/layoutStyles.css]:
```suggestion
  --textLink: #228be6; // OC BLUE 6
```

File: src/components/layoutStyles.css [hunk 5]
```diff
@@ -0,0 +1,40 @@
+/* global.css */
+
+body {
+  --bg: white;
+  --textNormal: #222;
+  --textTitle: #222;
```
Comment (lipis) [inline src/components/layoutStyles.css]:
```suggestion
  --textNormal: #212529; // OC GRAY 9
  --textTitle: #212529;
```

File: src/components/layoutStyles.css [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+  background-color: var(--bg);
+}
+
+body.dark {
+  -webkit-font-smoothing: antialiased;
+
+  --bg: #343a40;
+  --textNormal: #dbe4ff;
+  --textTitle: white;
```
Comment (lipis) [inline src/components/layoutStyles.css]:
```suggestion
  --textTitle: #fff;
```

File: src/components/layout.js (hunk 2 above)
Comment (mtliendo) [reply src/components/layout.js]: ah! That's what you meant, my bad!

File: src/components/layoutStyles.css [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+  --textLink: #228be6;
+  --hr: hsla(0, 0%, 0%, 0.2);
+
+  background-color: var(--bg);
+}
+
+body.dark {
+  -webkit-font-smoothing: antialiased;
+
+  --bg: #343a40;
```
Comment (lipis) [inline src/components/layoutStyles.css]:
```suggestion
  --bg: #212529;
```

Let's make it darker Grey 9

Comment (j-f1) [reply src/components/layoutStyles.css]: I think the macOS default background color is `#1e1e1e` for dark mode.

File: src/components/layoutStyles.css [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+  background-color: var(--bg);
+}
+
+body.dark {
+  -webkit-font-smoothing: antialiased;
+  /* OC Grey 9 */
+  --bg: #212529;
+  --textNormal: #dbe4ff;
+  --textTitle: #fff;
+  --textLink: #f03e3e;
```
Comment (lipis) [inline src/components/layoutStyles.css]: could we keep the link the same as with the light theme? and maybe adjust the blue to 5 or something..

Comment (lipis) [reply src/components/layoutStyles.css]: No need to change the link colors ideally.

Comment (mtliendo) [reply src/components/layoutStyles.css]: Sounds good, lemme send a SS real quick though

**Comment by Fausto95** [comment]
Can you inverse the order of the toggle please ?

 I expect it to be like: 
Capture d’écran 2020-03-23 à 15 52 40

**Comment by dwelle** [comment]
I agree with @Fausto95 - this also aligns it with Dan's blog.

File: src/components/layoutStyles.css (hunk 8 above)
Comment (mtliendo) [reply src/components/layoutStyles.css]: So this is if the dark theme links are kept as the light theme links currently are:
image

And this is when they are updated to the next darkest blue

image

I think the contrast for the links at the bottom may not be enough, thoughts?

Comment (dwelle) [reply src/components/layoutStyles.css]: Barely see a difference -- except maybe the bottom links, which are a bit too dark, true.

Comment (lipis) [reply src/components/layoutStyles.css]: If anything we need to go to the lightest blue or leave as is.

**Comment by j-f1** [comment]
Can you add `z-index: 1` to the toggle? Otherwise, the partially transparent text on the homepage overlaps it for some reason.

**Comment by mtliendo** [comment]
@j-f1 I can for sure add it, but can you send me a SS so I can better understand what you mean?

**Comment by j-f1** [comment]
Screenshot_2020-03-23 11 11 13

**Comment by mtliendo** [comment]
nice catch!

**Comment by lipis** [comment]
@j-f1 If nothing else.. go ahead and merge it!

**Comment by j-f1** [comment]
Oh — the “open Excalidraw” button needs to have its background colors changed in dark mode

**Comment by lipis** [comment]
Maybe in another iteration, as we need a white logo as well..

**Comment by lipis** [comment]
Thanks @mtliendo 🍻

**Comment by j-f1** [comment]
`filter: invert()` :)

**Comment by mtliendo** [comment]
Thanks for letting me tackle this one! Til next time 😁

**Pull request merged**